x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/perform_rule_upgrade/perform_rule_upgrade_route.schema.yaml

```diff
@@ -0,0 +1,865 @@
+openapi: 3.0.0
+info:
+  title: Perform Rule Upgrade API endpoint
+  version: '2023-10-31'
+paths:
+  /api/detection_engine/rules/prebuilt/_perform_upgrade:
+    post:
+      x-labels: [ess]
+      x-codegen-enabled: true
+      operationId: PerformRuleUpgrade
+      summary: Perform rule upgrade
+      description: Upgrade prebuilt detection rules.
+      tags:
+        - Rules API
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/UpgradeAllRulesRequest'
+                - $ref: '#/components/schemas/UpgradeSpecificRulesRequest'
+      responses:
+        200:
+          description: Indicates a successful call.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PerformRuleUpgradeResponseBody'
+
+components:
+  schemas:
+    PickVersionValues:
+      type: string
+      enum: [BASE, CURRENT, TARGET, MERGED]
+
+    RuleUpgradeSpecifier:
+      type: object
+      required:
+        - rule_id
+        - revision
+        - version
+      properties:
+        rule_id:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleSignatureId'
+        revision:
+          type: number
+        version:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleVersion'
+        pick_version:
+          $ref: '#/components/schemas/PickVersionValues'
```

Comment:
It would be super helpful to provide some description for other fields as well

Reply:
Yes, added to the fields here. What do you think about the description for revision?

```
Rule's current revision number. Should match the rule's revision number returned from the Review Rule Upgrade API endpoint.
```

I think that is understandable by the user (mentioning the upgrade endpoint), but is not technically correct. If a request slips in between calling /upgrade and /perform, the second request will fail because the revision number won't match. But that's hard to explain here.